services/code_hosting/git_hub_service.rb

```diff
@@ -1,131 +1,409 @@
 require_relative "code_hosting_service"
 require_relative "../../shared/logging_module"
+require_relative "../../shared/models/project"
+require_relative "../../fastfile-parser/fastfile_parser"
 
 require "set"
 require "octokit"
+require "git"
+require "addressable/uri"
+require "tty-command"
+require "securerandom"
+require "digest"
+require "pathname"
 
 module FastlaneCI
   # Data source that interacts with GitHub
   class GitHubService < CodeHostingService
     include FastlaneCI::Logging
 
     class << self
-      attr_accessor :status_context_prefix
-    end
+      attr_writer :temporary_git_storage
+
+      attr_accessor :temporary_storage_path
+
+      attr_writer :cache
```

Comment:
I don't like the idea of using caches right now. I think it's too early to optimize and it adds a ton of potential for things to go wrong.

Reply:
Justification below